src/MSA.BuildingBlocks.CosmosDbMigration/NoSQL/DatabaseMigration.cs

```diff
@@ -0,0 +1,222 @@
+﻿using Azure;
+using Microsoft.Azure.Cosmos;
+using MSA.BuildingBlocks.CosmosDbMigration.Abstractions;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Dynamic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MSA.BuildingBlocks.CosmosDbMigration.NoSQL;
+
+public class DatabaseMigration(CosmosClient cosmosClient, string databaseId, string containerId)
+    : BaseDatabaseMigration(cosmosClient, databaseId, containerId)
+{
+    public override async Task CloneContainer(string containerId, string partitionKey)
+    {
+        (IList<ExpandoObject> items, double requestCharge) = await GetItems().ConfigureAwait(false);
+
+        double createContainerAndUploadItemsCharge = await CreateContainerAndUploadItems(containerId, partitionKey, items).ConfigureAwait(false);
+        requestCharge += createContainerAndUploadItemsCharge;
+
+        Console.WriteLine($"{nameof(CloneContainer)} with items count {items.Count} operation cost {requestCharge} RUs.");
```

Comment:
To avoid excessive building and then refactoring, I recommend replacing Console.WriteLine statements with a more flexible logging framework, such as ILogger from Microsoft.Extensions.Logging.